iclr2017/python/get-submission-notes.py

```diff
@@ -0,0 +1,82 @@
+import argparse
+import openreview
+import csv
+
+#####################################################################################
+# This script produces a csv file with information on all notes associated with a paper
+# that includes the paper number, timestamp, note type, author, note content, pdf_url.
+#####################################################################################
+
+## Argument handling
+parser = argparse.ArgumentParser()
+parser.add_argument('--baseurl', help="base url")
+parser.add_argument('--username')
+parser.add_argument('--password')
+parser.add_argument('--ifile', help="input file name - default to submission_notes.csv")
+args = parser.parse_args()
+
+## Initialize the client library with username and password
+if args.username!=None and args.password!=None:
+    client = openreview.Client(baseurl=args.baseurl, username=args.username, password=args.password)
+else:
+    client = openreview.Client(baseurl=args.baseurl)
+## Initialize output file name
+file_name = "submission_notes.csv"
+if args.ifile!=None:
+    file_name = args.ifile
+
+submissions = client.get_notes(invitation='ICLR.cc/2017/conference/-/submission')
+allnotes = client.get_notes(invitation='ICLR.cc/2017/conference/-/paper.*/*')
+
+# paper_status[paper_num]['pdf'] = url to the submission pdf
+# paper_status[paper_num][note_id] dictionary including author, timestamp, type and content
+paper_status = {}
+# initialize paper_status for each submission
+for paper in submissions:
+    paper_status[paper.number] = {}
+    paper_status[paper.number]['pdf'] = paper.content['pdf']
+
+# add each note to the associated paper in paper_status
+for note in allnotes:
+    paper_num0 = note.invitation.split('paper')[1].split('/')[0]
+    note_type = note.invitation.split('paper' + str(paper_num0))[1]
+    # WHY does paper_num work and paper_num0 not?
+    paper_num = int(note.invitation.split('paper')[1].split(note_type)[0])
+    if paper_num in paper_status:
+        author = note.writers[0]
```

Comment:
It is better to use the first element of signatures instead of writers. There are some case they are not the same, and signatures is how the use signed the note.